tests/testthat/test_maraca.R

```diff
@@ -189,12 +164,103 @@ test_that("Maraca wrong params", {
       c(
         outcome = "GROUP", arm = "TRTP",
         value = "AVAL0"
-      ), 12
+      ), fixed_followup_days
     ),
     regexp = "The outcome column must be characters"
   )
 })
 
+test_that("Maraca printing", {
+  file <- fixture_path("hce_scenario_c.csv")
+  data <- read.csv(file, stringsAsFactors = FALSE)
+  tte_outcomes <- c(
+    "Outcome I", "Outcome II", "Outcome III", "Outcome IV"
+  )
+  continuous_outcome <- "Continuous outcome"
+  arm_levels <- c(active = "Active", control = "Control")
+  column_names <- c(
+    outcome = "GROUP", arm = "TRTP", value = "AVAL0"
+  )
+  mar <- maraca(
+    data, tte_outcomes, continuous_outcome, arm_levels, column_names, 3 * 365,
+    compute_win_odds = TRUE
+  )
+  data_rf <- .reformat_and_check_data(data, tte_outcomes, continuous_outcome,
+                                   arm_levels, column_names = column_names
+  )
+  win_odds <- .compute_win_odds(data_rf)
+
+  mar_no_win_odds <- maraca(
+    data, tte_outcomes, continuous_outcome, arm_levels, column_names, 3 * 365,
+    compute_win_odds = FALSE
+  )
+
+  data$AVAL0[[3]] <- NA
+  mar_na <- maraca(
+    data, tte_outcomes, continuous_outcome, arm_levels, column_names, 3 * 365,
+    compute_win_odds = TRUE
+  )
+  data_rf <- .reformat_and_check_data(data[!is.na(data$AVAL0), ], tte_outcomes,
+                                      continuous_outcome,
+                                      arm_levels, column_names = column_names
+  )
+  win_odds_na <- .compute_win_odds(data_rf)
+
+  expect_output(print(mar),
```

Comment:
In this case, expect output is not a good option. What you really want is something like this:

```
# Function to compare long text line by line, organised as a vector of strings.
expect_text_equal <- function(result, expected) {
  expect_equal(length(result), length(expected))

  for (i in seq_along(expected)) {
    expect_equal(result[[i]], expected[[i]])
  }

}
```

And pass the capture.output() of the print and everything you expect as a vector of strings.

Reply:
Thanks, very helpful, I updated to using your function now.